test/qr.jl

```diff
@@ -1,223 +1,64 @@
             end
         end
-        qr_full!(copy!(Ac, A), (Q2, noR); positive = true, pivoted = true)
-        @test Q == Q2
+    else
+        if T ∈ BLASFloats
+            TestSuite.test_qr(T, (m, n))
+            LAPACK_QR_ALGS = (
+                LAPACK_HouseholderQR(; positive = false, pivoted = false, blocksize = 1),
+                LAPACK_HouseholderQR(; positive = false, pivoted = false, blocksize = 2),
+                LAPACK_HouseholderQR(; positive = false, pivoted = true, blocksize = 1),
+                #LAPACK_HouseholderQR(; positive=false, pivoted=true, blocksize=2), # not supported
+                LAPACK_HouseholderQR(; positive = true, pivoted = false, blocksize = 1),
+                LAPACK_HouseholderQR(; positive = true, pivoted = false, blocksize = 2),
+                LAPACK_HouseholderQR(; positive = true, pivoted = true, blocksize = 1),
+                #LAPACK_HouseholderQR(; positive=true, pivoted=true, blocksize=2), # not supported
+            )
+            TestSuite.test_qr_algs(T, (m, n), LAPACK_QR_ALGS)
+        elseif T ∈ GenericFloats
+            TestSuite.test_qr(T, (m, n); test_null = false, test_pivoted = false, test_blocksize = false)
+            GLA_QR_ALGS = (GLA_HouseholderQR(),)
+            TestSuite.test_qr_algs(T, (m, n), GLA_QR_ALGS; test_null = false)
+        end
     end
 end
-
-@testset "qr_compact, qr_full and qr_null for Diagonal{$T}" for T in (BLASFloats..., GenericFloats...)
-    rng = StableRNG(123)
-    atol = eps(real(T))^(3 / 4)
-    for m in (54, 0)
-        Ad = randn(rng, T, m)
-        A = Diagonal(Ad)
-
-        # compact
-        Q, R = @constinferred qr_compact(A)
-        @test Q isa Diagonal{T} && size(Q) == (m, m)
-        @test R isa Diagonal{T} && size(R) == (m, m)
-        @test Q * R ≈ A
-        @test isunitary(Q)
-
-        # compact and positive
-        Qp, Rp = @constinferred qr_compact(A; positive = true)
-        @test Qp isa Diagonal{T} && size(Qp) == (m, m)
-        @test Rp isa Diagonal{T} && size(Rp) == (m, m)
-        @test Qp * Rp ≈ A
-        @test isunitary(Qp)
-        @test all(≥(zero(real(T))), real(diag(Rp))) &&
-            all(≈(zero(real(T)); atol), imag(diag(Rp)))
-
-        # full
-        Q, R = @constinferred qr_full(A)
-        @test Q isa Diagonal{T} && size(Q) == (m, m)
-        @test R isa Diagonal{T} && size(R) == (m, m)
-        @test Q * R ≈ A
-        @test isunitary(Q)
-
-        # full and positive
-        Qp, Rp = @constinferred qr_full(A; positive = true)
-        @test Qp isa Diagonal{T} && size(Qp) == (m, m)
-        @test Rp isa Diagonal{T} && size(Rp) == (m, m)
-        @test Qp * Rp ≈ A
-        @test isunitary(Qp)
-        @test all(≥(zero(real(T))), real(diag(Rp))) &&
-            all(≈(zero(real(T)); atol), imag(diag(Rp)))
-
-        # null
-        N = @constinferred qr_null(A)
-        @test N isa AbstractMatrix{T} && size(N) == (m, 0)
+if !is_buildkite
+    for T in (BLASFloats..., GenericFloats...)
```

Comment:
Could the block below just be below the `else # if isbuildkite` on line 37, before the `if T ∈ BLASFloats`?
(Same applies for lq.jl)